File: Low Level Design (LLD)/Problems/ParkingLot/Java/model/Ticket.java
package model;

import java.time.LocalDateTime;

public class Ticket {
    private final Vehicle vehicle;
    private final ParkingSlot slot;
    private final LocalDateTime entryTime;

    public Ticket(Vehicle vehicle, ParkingSlot slot) {
        this.vehicle = vehicle;
        this.slot = slot;
        this.entryTime = LocalDateTime.now();
    }

    public Vehicle getVehicle() {
        return vehicle;
    }

    public ParkingSlot getSlot() {
        return slot;
    }

    public LocalDateTime getEntryTime() {
        return entryTime;
    }
}

File: Low Level Design (LLD)/Problems/02. ParkingLot/Cpp/model/src/Ticket.cpp
#include "../include/Ticket.h"

Ticket::Ticket(Vehicle* vehicle, ParkingSlot* parkingSlot, std::chrono::system_clock::time_point entryTime) 
    : vehicle(vehicle), parkingSlot(parkingSlot), entryTime(std::chrono::system_clock::now()) {}

Vehicle* Ticket::getVehicle() const {
    return vehicle;
}

ParkingSlot* Ticket::getSlot() const {
    return parkingSlot;
}

std::chrono::system_clock::time_point Ticket::getEntryTime() const {
    return entryTime;
}
